hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -485,9 +491,21 @@ private boolean checkContainersReplicatedOnNode(TrackedNode dn)
           unclosed, unClosedIDs.stream().map(
               Object::toString).collect(Collectors.joining(", ")));
     }
+    dn.setContainersReplicatedOnNode(underReplicatedIDs, unClosedIDs);
     return underReplicated == 0 && unclosed == 0;
   }
 
+  public Map<String, List<ContainerID>> getContainersReplicatedOnNode(TrackedNode dn) {
+    Iterator<TrackedNode> iterator = trackedNodes.iterator();
```

Comment:
I don't think you need to iterator here. Can you just do `trackedNodes.get(dn)`, as the `dn` is a TrackedNode.

Reply:
Ah, you are correct. Sorry!

Reply:
We are storing trackedNodes as Set<TrackedNodes>, not Map. I don't think we can do a simple get.